cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1675,18 +1709,27 @@ def request_integration():
 
 
 @when_none('endpoint.aws.joined',
-           'endpoint.gcp.joined')
+           'endpoint.gcp.joined',
+           'endpoint.openstack.joined',
+           'endpoint.vsphere.joined',
+           'endpoint.azure.joined')
 @when('kubernetes-master.cloud-request-sent')
 def clear_requested_integration():
+    remove_state('kubernetes-master.cloud.pending')
     remove_state('kubernetes-master.cloud-request-sent')
+    remove_state('kubernetes-master.cloud.blocked')
+    remove_state('kubernetes-master.cloud.ready')
 
 
 @when_any('endpoint.aws.ready',
           'endpoint.gcp.ready',
           'endpoint.openstack.ready',
-          'endpoint.vsphere.ready')
-@when_not('kubernetes-master.restarted-for-cloud')
-def restart_for_cloud():
+          'endpoint.vsphere.ready',
+          'endpoint.azure.ready')
+@when_not('kubernetes-master.cloud.blocked',
+          'kubernetes-master.cloud.ready',
+          'kubernetes-master.restarted-for-cloud')  # compat. TODO: remove
```

Comment:
Might be worthwhile to handle the migration from `kubernetes-master.restarted-for-cloud` to `kubernetes-master.cloud.ready` in the `upgrade-charm` hook so you don't have to do this.